### PR TITLE
feat: adding vertexai instrumentation support

### PR DESCRIPTION
**Pull request by darshit-s3**
# Description

Adding support for Google Vertex AI instrumentation for methods:
- generateContent
- generateContentStream
- sendMessage
- sendMessageStream

Reference PR for trace-attributes: https://github.com/Scale3-Labs/langtrace-trace-attributes/pull/87

# Checklist for adding new integration:

- [ ] Defined `APIS` in constants [folder](../src/constants/instrumentation/).
- [ ] Updated `SERVICE_PROVIDERS` in [common.ts](../src/constants/common.ts)
- [ ] Created a folder under [instrumentation](../src/instrumentation/) with the name of the integration with atleast `patch.ts` and `instrumentation.ts` files.
- [ ] Added instrumentation in `allInstrumentations` in [init.ts](../src/init/init.ts) and to the `InstrumentationType` in [types.ts](../src/init/types.ts) files.
- [ ] Added examples for the new integration in the [examples](../src/examples/) folder.
- [ ] Updated [package.json](../package.json) to install new dependencies for devDependencies.
- [ ] Updated the [README.md](../README.md) of [langtrace-typescript-sdk](https://github.com/Scale3-Labs/langtrace-typescript-sdk) to include the new integration in the supported integrations table.
- [ ] Updated the [README.md](https://github.com/Scale3-Labs/langtrace?tab=readme-ov-file#supported-integrations) of Langtrace's [repository](https://github.com/Scale3-Labs/langtrace) to include the new integration in the supported integrations table.
- [ ] Added new integration page to supported integrations in [Langtrace Docs](https://github.com/Scale3-Labs/langtrace-docs)

### CODE REVIEW

File: src/instrumentation/vercel/patch.ts [hunk 1]
```diff
@@ -14,6 +14,7 @@ export function generateTextPatch (
   version?: string
 ): (...args: any[]) => any {
   const patchThis = this
+
```
Comment (rohit-kadhe) [inline src/instrumentation/vercel/patch.ts:17]: nit revert back not relevant to this pr

File: src/instrumentation/vertexai/patch.ts [hunk 2]
```diff
@@ -0,0 +1,241 @@
...
+  text?: any
+  candidates: Candidate[]
+  usageMetadata: UsageMetadata
+}
+
+interface Resp {
+  stream?: any
+  response: Response
+}
+
```
Comment (rohit-kadhe) [inline src/instrumentation/vertexai/patch.ts:70]: Create a types.ts file in the same folder and move these there. You can look at `cohere` for an example

**Review by rohit-kadhe — CHANGES_REQUESTED** [review]

File: src/instrumentation/vercel/patch.ts (hunk 1 above)
Comment (darshit-s3) [reply src/instrumentation/vercel/patch.ts]: done

File: src/instrumentation/vertexai/patch.ts (hunk 2 above)
Comment (darshit-s3) [reply src/instrumentation/vertexai/patch.ts]: moved

**Review by karthikscale3 — APPROVED** [review]

**Review by rohit-kadhe — APPROVED** [review]

**Pull request merged**